pkg/curatedpackages/kubeclient.go

```diff
@@ -0,0 +1,57 @@
+package curatedpackages
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	packagesv1 "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+)
+
+// NewKubeClientFromFilename creates a controller-runtime k8s client for use
+// by CLI commands.
+//
+// If the RESTConfigurator is nil, a default configurator will be used.
+func NewKubeClientFromFilename(kubeConfigFilename string, rc RESTConfigurator) (client.Client, error) {
```

Comment:
I assume this would be used from factory.go? Currently any bootstrapping is performed from factory.go.

Reply:
I'm not sure that I understand your question. The function is exported, so you can call it from anywhere you like (so long as you import the curatedpackages package). I suspect there might be a better place for this to live long term, but for now, this seems to be fine.